python/ray/dag/compiled_dag_node.py

```diff
@@ -108,11 +108,16 @@ def do_exec_tasks(
             task.prepare()
 
         done = False
+        # Cache to store channel results for an actor loop iteration.
+        cache: Dict[ChannelInterface, Any] = {}
         while True:
             if done:
                 break
+            cache.clear()
```

Comment:
Can we update / clear the cache during the DAG execution? Otherwise we may leak memory especially for GPU tensors. We can also unify this codepath with the ref counting that is currently done for intra-process channels.

Reply:
Done.